base-cert-manager/client.yaml

```diff
@@ -0,0 +1,61 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: &app cockroachdb-client
+  annotations:
+    "app.uw.systems/description": "Used to connect and query the cockroachdb databases."
+    "app.uw.systems/repos.cockroachdb-manifests": "https://github.com/utilitywarehouse/cockroachdb-manifests"
+    "app.uw.systems/tier": "tier_4"
+    "app.uw.systems/tags.oss": "true"
+    "secret.reloader.stakater.com/reload": "cockroachdb-ca, cockroachdb.client.root, cockroachdb.node"
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: *app
+  template:
+    metadata:
+      name: *app
+      labels:
+        app: *app
+    spec:
+      terminationGracePeriodSeconds: 60
+      containers:
+        - name: cockroachdb-client
+          image: cockroachdb
+          command:
+            - "/bin/sh"
+            - "-c"
+            - |
+              sleep infinity & PID=$!
+              trap "kill $PID" TERM
+              wait
+              echo "shutting down..."
+          env:
+            - name: COCKROACH_HOST
+              valueFrom:
+                configMapKeyRef:
+                  name: cockroach
+                  key: cockroach.host
+            - name: COCKROACH_PORT
+              valueFrom:
+                configMapKeyRef:
+                  name: cockroach
+                  key: cockroach.port
+            - name: COCKROACH_CERTS_DIR
+              value: "/cockroach/cockroach-certs"
+          resources:
+            requests:
+              cpu: 0m
+              memory: 16Mi
+            limits:
+              cpu: 1000m
+              memory: 512Mi
+          volumeMounts:
+            - name: client-certs
+              mountPath: /cockroach/cockroach-certs
+      volumes:
+        - name: client-certs
+          secret:
+            secretName: cockroachdb.client.root
+            defaultMode: 256
```

Comment:
curious: why are file modes written in base 10? I'm almost universally used to them being written in octal

Reply:
I believe I tried octal initially and that didn't work here

Reply:
> I believe I tried octal initially and that didn't work here

I went down a rabbit hole to realise:

* YAML 1.1: octal is e.g. `0446` https://yaml.org/type/int.html
* YAML 1.2: octal is e.g. `0o446` https://yaml.org/spec/1.2.2/#24-tags
* kubernetes YAML support is not clearly defined (it depends on how its dependencies support it https://github.com/kubernetes/kubernetes/issues/34146)

what a mess

Reply:
> what a mess

Preach!